source/Government.cpp

```diff
@@ -316,11 +343,27 @@ double Government::InitialPlayerReputation() const
 // Get the amount that your reputation changes for the given offense.
 double Government::PenaltyFor(int eventType) const
 {
-	double penalty = 0.;
-	for(const auto &it : penaltyFor)
-		if(eventType & it.first)
-			penalty += it.second;
-	return penalty;
+	return PenaltyHelper(eventType, penaltyFor);
+}
+
+
+
+double Government::PenaltyFor(int eventType, const Government *other) const
+{
+	if(!other)
+		return 0.;
+
+	if(other == this)
+		return PenaltyFor(eventType);
```

Comment:
do we really need that other PenaltyFor? cant we just do that one line here? (and also request a ref and not a pointer for the gov, or consider nullptr as this gov?)

Reply:
Getting a reference requires dereferencing when we don't necessarily need to.
I don't know why we would assume it is `this` when passed a nullpointer, especially when it can potentially just be `this`.

Reply:
I mean its fine its just that we dont use that function for anything else

Reply:
We don't. I deleted that one now.